hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -2646,6 +2655,7 @@ public void run() {
       } catch (Throwable t) {
         LOG.error("Failed to transfer block {}", b, t);
       } finally {
+        transferringBlock.remove(b);
```

Comment:
Are we sure we are cleaning this up and won't leave garbage?

Reply:
I don't catch you advice, transferringBlock.add and transferringBlock.remove could be only called DataTransfer.run, I think won't leave garbage.